wtxmgr/tx_test.go

```diff
@@ -1496,85 +1533,122 @@ func TestInsertConfirmedDoubleSpendTx(t *testing.T) {
 		Block: Block{Height: 100},
 		Time:  time.Now(),
 	}
-
-	cb := newCoinBase(1e8)
-	cbRec, err := NewTxRecordFromMsgTx(cb, b100.Time)
+	cb1 := newCoinBase(1e8)
+	cbRec1, err := NewTxRecordFromMsgTx(cb1, b100.Time)
 	if err != nil {
 		t.Fatal(err)
 	}
-	if err := store.InsertTx(ns, cbRec, &b100); err != nil {
-		t.Fatal(err)
-	}
+	commitDBTx(t, store, db, func(ns walletdb.ReadWriteBucket) {
+		if err := store.InsertTx(ns, cbRec1, &b100); err != nil {
+			t.Fatal(err)
+		}
+	})
 
-	// Then, we'll create three spends spends from the same coinbase output.
-	// The first two will remain unconfirmed, while the last should confirm
-	// and remove the remaning unconfirmed from the wallet's store.
-	firstSpend := spendOutput(&cbRec.Hash, 0, 5e7, 5e7)
-	firstSpendRec, err := NewTxRecordFromMsgTx(firstSpend, time.Now())
+	// Then, we'll create three spends from the same coinbase output. The
+	// first two will remain unconfirmed, while the last should confirm and
+	// remove the remaining unconfirmed from the wallet's store.
+	firstSpend1 := spendOutput(&cbRec1.Hash, 0, 5e7, 5e7)
+	firstSpendRec1, err := NewTxRecordFromMsgTx(firstSpend1, time.Now())
 	if err != nil {
 		t.Fatal(err)
 	}
-	if err := store.InsertTx(ns, firstSpendRec, nil); err != nil {
+	commitDBTx(t, store, db, func(ns walletdb.ReadWriteBucket) {
+		if err := store.InsertTx(ns, firstSpendRec1, nil); err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	secondSpend1 := spendOutput(&cbRec1.Hash, 0, 4e7, 6e7)
+	secondSpendRec1, err := NewTxRecordFromMsgTx(secondSpend1, time.Now())
+	if err != nil {
 		t.Fatal(err)
 	}
-	secondSpend := spendOutput(&cbRec.Hash, 0, 4e7, 6e7)
-	secondSpendRec, err := NewTxRecordFromMsgTx(secondSpend, time.Now())
+	commitDBTx(t, store, db, func(ns walletdb.ReadWriteBucket) {
+		if err := store.InsertTx(ns, secondSpendRec1, nil); err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	// We'll also create another output and have one unconfirmed and one
+	// confirmed spending transaction also spend it.
+	cb2 := newCoinBase(2e8)
+	cbRec2, err := NewTxRecordFromMsgTx(cb2, b100.Time)
 	if err != nil {
 		t.Fatal(err)
 	}
-	if err := store.InsertTx(ns, secondSpendRec, nil); err != nil {
+	commitDBTx(t, store, db, func(ns walletdb.ReadWriteBucket) {
+		if err := store.InsertTx(ns, cbRec2, &b100); err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	firstSpend2 := spendOutput(&cbRec2.Hash, 0, 5e7, 5e7)
+	firstSpendRec2, err := NewTxRecordFromMsgTx(firstSpend2, time.Now())
+	if err != nil {
 		t.Fatal(err)
 	}
+	commitDBTx(t, store, db, func(ns walletdb.ReadWriteBucket) {
+		if err := store.InsertTx(ns, firstSpendRec2, nil); err != nil {
+			t.Fatal(err)
+		}
+	})
 
 	// At this point, we should see both unconfirmed transactions within the
 	// store.
-	unminedTxs, err := store.UnminedTxs(ns)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if len(unminedTxs) != 2 {
-		t.Fatalf("expected 2 unmined txs, got %d", len(unminedTxs))
-	}
+	commitDBTx(t, store, db, func(ns walletdb.ReadWriteBucket) {
```

Comment:
update comments to reflect the change.

Reply:
Are you referring to the individual database transactions? A comment was included at the top of the test indicating that a new database transaction would be used for each interaction.